pkgs/native_assets_cli/lib/src/config.dart

```diff
@@ -404,6 +406,168 @@ class BuildOutputBuilder extends HookOutputBuilder {
       syntax.BuildOutput.fromJson(super._syntax.json);
 }
 
+extension DataAssetsDirectoryExtension on BuildOutputBuilder {
+  /// Extension on [BuildOutput] to handle data asset directories and files.
+  ///
+  /// This extension provides a convenient way for build hooks to add
+  /// [DataAsset] dependencies from one or more directories or individual files.
+  ///
+  /// If any specified path does not exist, a [FileSystemException] is thrown.
+  /// Any error during the directory listing is caught and rethrown with
+  /// additional context.
+  Future<void> addDataAssetDirectories(
+    List<String> paths, {
+    required BuildInput input,
+  }) async {
+    final packageName = input.packageName;
+    final packageRoot = input.packageRoot;
+    for (final path in paths) {
+      final resolvedUri = packageRoot.resolve('$packageName/$path');
+      final directory = Directory.fromUri(resolvedUri);
+      final file = File.fromUri(resolvedUri);
+
+      if (await directory.exists()) {
+        // Process as a directory.
+        addDependency(directory.uri);
+        try {
+          await for (final entity in directory.list(
+            recursive: true,
+            followLinks: false,
+          )) {
+            // Add dependency for every file and directory found.
+            addDependency(entity.uri);
+          }
+        } on FileSystemException catch (e) {
+          throw FileSystemException(
+            'Error reading directory "$path": ${e.message}',
+            directory.path,
+            e.osError,
+          );
+        }
+      } else if (await file.exists()) {
+        addDependency(file.parent.uri);
+        addDependency(file.uri);
+      } else {
+        throw FileSystemException(
+          'Path does not exist',
+          resolvedUri.toFilePath(windows: Platform.isWindows),
+        );
+      }
+    }
+  }
+
+  void addDependency(Uri uri) {
+    final dependencies = _syntax.dependencies ?? [];
```

Comment:
Why access the syntax? You should be able to just use the addDependency Dart API.

Reply:
Ah I started this as an extension on BuildOutput and I don't think I had access to addDependency there, pushed a fix.